### ISSUE TITLE
ChatEventPriority is messed up

### ISSUE THREAD

**Issue by GitNether** [festino-mc-plugins/ClickableLinks#8]
Messes with my HoverStats plugin, which should have the highest priority (monitor)
![image](https://github.com/festino-mc-plugins/ClickableLinks/assets/100975664/8e585fea-cf3d-40e9-b6a4-3769e63ff235)

**Comment by festino** [comment]
I suspect that this has nothing to do with event priority: ClickableLinks plugin completely cancels the event and sends the message itself via tellraw.

I'll look into how clickEvent and hoverEvent are added in other plugins (specifically HoverStats, but that seems to be closed source).

**Comment by festino** [comment]
HoverStats plugin uses `Player#spigot().sendMessage()` to send a formatted message and `AsyncPlayerChatEvent#getRecipients().clear()` to cancel the event. ClickableLinks does almost the same thing, so I think the plugins may not be compitable.

Did they work well together before the latest updates (v2.0.0)?

**Comment by festino** [comment]
If that's the problem, the only thing I can think of is to intercept the HoverStats result later (for example, as a packet sent to the player) and edit it.

This is highly undesirable, as it will complicate plugin support (possible version dependency) and may have a bad effect on performance. Although both problems may turn out to be insignificant in practice.

**Comment by GitNether** [comment]
Do you think it would be possible to extract any existing hover effects etc. of the message and re-add them, before sending your message? 
The deluxetag is also iffy as you can see.

**Comment by festino** [comment]
Yes, I am completely confident that this is possible. I can try to add such an option, as if the priority is higher than MONITOR. However, multiple plugins with this “priority” could no longer work simultaneously, I think.

I don't quite understand what you mean by "iffy".

I also want to say that I have not yet tested the interaction between ClickableLinks and HoverStats. Everything said above is theoretical, based on the decompiled HoverStats code.

**Comment by festino** [comment]
I think that I should not use NMS due to the significant complexity of maintenance. I would have to intercept packets, interfering with the netty pipeline, and also use some magic with reflection. I can only suggest trying the option with a dependency on ProtocolLib (never used its API before :D).

**Comment by festino** [comment]
The only problem with ProtocolLib is that the plugin receives all system messages and I need to determine if it is actually a player chat message. The current idea is this: monitor the AsyncPlayerChatEvent with MONITOR priority, and process the message only when the content of the message matches. But this will not work if, for example, there is a censorship plugin that will replace swear words with asterisks.

**Comment by festino** [comment]
Hope you are still using the plugin. I just published a new release:
https://github.com/festino-mc-plugins/ClickableLinks/releases/tag/v2.1.0

It adds a new configuration option `modify-packets`. Requires ProtocolLib. The plugin will add click and hover events to messages when `modify-packets: true`. It may be buggy, so I hope for any feedback.

**Comment by GitNether** [comment]
Hey hey, I am currently not working alot on the server due to work and other projects etc.
I'll make sure to give it a test this weekend at latest ^^
